rosbridge_library/src/rosbridge_library/internal/services.py

```diff
@@ -120,9 +122,14 @@ def call_service(node_handle, service, args=None):
     # Populate the instance with the provided args
     args_to_service_request_instance(service, inst, args)
 
-    client = node_handle.create_client(service_class, service)
+    if service_class in clients:
+        client = clients[service_class]
+    else:
+        client = node_handle.create_client(service_class, service)
+        clients[service_class] = client
```

Comment:
This does not work if you have multiple services with the same service type but different names. Let's say you have two identical services under a different namespace:
- `/robot_1/do_something`
- `/robot_2/do_something`

You will only call one of these services (the one you called first), as they have the same service type.

Reply:
I'll consider it, thanks!

Reply:
@achim-k 
> I would propose to simply [destroy](https://docs.ros2.org/galactic/api/rclpy/api/services.html#rclpy.service.Service.destroy) the service client after the call.

Calling `rclpy.service.Service.destroy` is user responsibility, so I think it cannot be used in the library.
And I know that `node_handle.destroy_client(client)` does not free all memory.

Reply:
@achim-k https://github.com/RobotWebTools/rosbridge_suite/pull/774/commits/1909996867cc757253030734a27f7c1c1d2378b5 should work.